packages/core/src/conversation/ConversationService/ConversationService.ts

```diff
@@ -110,6 +116,10 @@ export class ConversationService extends TypedEventEmitter<Events> {
       this.mlsService.on(MLSServiceEvents.MLS_EVENT_DISTRIBUTED, data => {
         this.emit(MLSServiceEvents.MLS_EVENT_DISTRIBUTED, data);
       });
+      this.mlsService.on(MLSServiceEvents.KEY_MATERIAL_UPDATE_FAILURE, ({error, groupId}) => {
```

Comment:
is it possible that two KEY_MATERIAL_UPDATE_FAILURE events for same group arrive concurrently then won't we duplicate all the steps inside reactToKeyMaterialUpdateFailure? shouldn't we check if a recovery is in progress then we return that promise instead of creating new one(maintaining a Map)?